src/triage/component/catwalk/baselines/thresholders.py

```diff
@@ -0,0 +1,184 @@
+import numpy as np
+import pandas as pd
+from six import string_types
+
+from triage.exceptions import BaselineFeatureNotInMatrix
+
+REQUIRED_KEYS = ['feature_name', 'operator', 'threshold']
+
+
+class SimpleThresholder(object):
+    """ The simple thresholder applies a set of predetermined logical rules to a
+    test matrix to classify entities. By default, it will classify entities as 1
+    if they satisfy any of the rules. When 'or' is set as the logical_operator
+    attribute, it will classify entities as 1 if they pass *all* of the rules.
+
+    Rules are passed as either strings in the format 'x1 > 5' or dictionaries in
+    the format {feature_name: 'x1', operator: '>', threshold: 5}. The
+    feature_name, operator, and threshold keys are required. Eventually, this
+    class may be abstracted into a BaseThreshold class and more complicated
+    thresholders could be built around new keys in the dictionaries (e.g., by
+    specifying scores that could be applied (and possibly summed) to entities
+    satisfying rules) or by an alternative dictionary format that specifies
+    more complicated structures for applying rules (for example:
+        {
+            or: [
+                {or: [{}, {}]},
+                {and: [{}, {}]}
+            ]
+        }
+    where rules and operators that combine them can be nested).
+    """
+    def __init__(self, rules, logical_operator='or'):
+        self.rules = rules
```

Comment:
nice use of `property`!

if you want to get fancier, I believe you can avoid the extra, "hidden" attribute, `_rules`, and avoid recursion by internally accessing the data directly:

    @property
    def rules(self):
        return vars(self)['rules']

    @rules.setter
    def rules(self, rules):
        vars(self)['rules'] = …

Reply:
…NICE!